python_modules/dagster/dagster/_utils/alert.py

```diff
@@ -211,13 +216,15 @@ def email_on_run_failure(context: RunFailureSensorContext):
 
         if smtp_type == "SSL":
             send_email_via_ssl(
-                email_from, email_password, email_to, message, smtp_host, smtp_port=smtp_port or 465
+                email_from, email_user or email_from, email_password, email_to, message, smtp_host, smtp_port=smtp_port or 465
```

Comment:
can `email_user` and `email_from` be supplied together or are they mutually exclusive? if the latter, i'd suggest having a check at the beginning of this function to guard the disallowed case. example code elsewhere in the codebase: https://github.com/dagster-io/dagster/blob/9d4eeeff48147111a71c713c31c7882d650997b8/python_modules/dagster/dagster/_core/definitions/reconstruct.py#L273-L276


besides, `email_user or email_from` will easily get messy when the params grow. so after checks (if needed), we could convert them to a single variable just to make sure the user-provided value can be properly traveled together through the code path.

Reply:
Yes, both email_user and email_from can be different and are sometimes both required.